docs/api-reference/io-columns-tree-featurization-binary-classification.md

```diff
@@ -0,0 +1,14 @@
+### Input and Output Columns
+The input label column data must be <xref:System.Boolean>.
+The input features column data must be a known-sized vector of <xref:System.Single>.
+
+This estimator outputs the following columns:
+
+| Output Column Name | Column Type | Description|
+| -- | -- | -- |
+| `Trees` | Vector of <xref:System.Single> | The output values of all trees. |
```

Comment:
>The output values of all trees [](start = 45, length = 30)

One value per tree I would assume?

Reply:
Yes, details are included in `tree-featurization-prediction.md`
```md
### Prediction Details
This estimator produces several output columns from a tree ensemble model. Assume that the model contains only one decision tree:

                   Node 0
                   /    \
                 /        \
               /            \
             /                \
           Node 1            Node 2
           /    \            /    \
         /        \        /        \
       /            \     Leaf -3  Node 3
      Leaf -1      Leaf -2         /    \
                                 /        \
                                Leaf -4  Leaf -5

Assume that the input feature vector falls into `Leaf -1`. The output `Trees` may be a 1-element vector where
the only value is the decision value carried by `Leaf -1`. The output `Leaves` is a 0-1 vector. If the reached
leaf is the $i$-th (indexed by $-(i+1)$ so the first leaf is `Leaf -1`) leaf in the tree, the $i$-th value in `Leaves`
would be 1 and all other values would be 0. The output `Paths` is a 0-1 representation of the nodes passed
through before reaching the leaf. The $i$-th element in `Paths` indicates if the $i$-th node (indexed by $i$) is touched.
For example, reaching `Leaf -1` lead to $[1, 1, 0, 0]$ as the `Paths`. If there are multiple trees, this estimator
just concatenates `Trees`'s, `Leaves`'s, `Paths`'s from all trees (first tree's information comes first in the concatenated vectors).

Check the See Also section for links to usage examples.
```

Reply:
Yes. I will extend this string a bit.

---
In reply to: [291364613](https://github.com/dotnet/machinelearning/pull/3812#discussion_r291364613) [](ancestors = 291364613)